src/features/stats-widget/StatsWidget.tsx

```diff
@@ -0,0 +1,110 @@
+//- React Imports
+import type { FC } from 'react';
+
+//- Library Imports
+import { ethers } from 'ethers';
+
+//- Hook Imports
+import { useDomainMetrics } from '../../lib/hooks/useDomainMetrics';
+import { formatEthers, formatNumber } from '../../lib/util/number/number';
+import { usePaymentTokenInfo } from '../../lib/hooks/usePaymentTokenInfo';
+
+type StatsWidgetProps = {
+	domainId: string;
+	isNFTView: boolean;
+};
+
+const StatsWidget: FC<StatsWidgetProps> = ({ domainId, isNFTView }) => {
```

Comment:
I think we can change this component to be a bit more reusable by reducing what it's responsible for.

At the moment, it's firing some queries, and turning the return into stat widgets. I think it would be cleaner if this component was just responsible for turning a list of stats into Stat Widgets.

To achieve this, we would have to change the `StatWidget` component to take in an array of `Stat`, and the parent would be responsible for feeding it the correct data. This also makes it much easier to test because we just need to assert that the data sent to it is rendered correctly, rather than having to mock hooks.

[Here's an example of what I did in zApp Staking](https://github.com/zer0-os/zApp-Staking/blob/feat/staking-v1/src/features/my-deposits/DepositsTable.tsx). Looking at my code, I think it actually would make more sense to put the stat logic in the page (in my case, Deposits) rather than the DepositTable, but you get the idea.